src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -155,18 +162,19 @@ class RelaxedRigidContactsState(ContactsState):
     def __eq__(self, other: RelaxedRigidContactsState) -> bool:
         return hash(self) == hash(other)
 
-    @staticmethod
-    def build() -> RelaxedRigidContactsState:
+    @classmethod
+    def build(cls: type[Self]) -> Self:
         """Create a `RelaxedRigidContactsState` instance"""
 
         return RelaxedRigidContactsState()
```

Comment:
At this point we could directly use `cls`
```suggestion
        return cls()
```